torch_geometric/utils/hetero.py

```diff
@@ -47,3 +47,15 @@ def get_unused_node_types(node_types: List[NodeType],
                           edge_types: List[EdgeType]) -> Set[NodeType]:
     dst_node_types = set(edge_type[-1] for edge_type in edge_types)
     return set(node_types) - set(dst_node_types)
+
+
+def check_add_self_loops(module: torch.nn.Module, edge_types: List[EdgeType]):
```

Comment:
putting the edges check here doesn't feel quite right, but it keeps the check quite easy to reuse in the three different places - happy to move it